src/platform/site-wide/user-nav/components/SearchTypeAhead.jsx

```diff
@@ -0,0 +1,133 @@
+
+    this.setState({ suggestions });
+  };
+
+  optionClasses = selected => classNames('dropdown-option', { selected });
+
+  render() {
+    return (
+      <form
+        ref="searchform"
+        acceptCharset="UTF-8"
+        id="search"
+        method="get"
+        action="/search/"
+      >
+        <Downshift
+          onChange={this.handleChange}
+          onInputValueChange={this.onInputValueChange}
+          inputValue={this.state.input}
+          itemToString={suggestion => suggestion}
+        >
+          {({
+            getInputProps,
+            getItemProps,
+            getLabelProps,
+            isOpen,
+            inputValue,
+            highlightedIndex,
+            selectedItem,
+          }) => (
+            <div>
+              <label
+                {...getLabelProps()}
+                className="sr-only"
+                htmlFor="service-type-ahead-input"
+              >
+                Search{' '}
+              </label>
+              <span id="service-typeahead">
+                <div className="va-flex">
+                  <input
+                    {...getInputProps({
+                      placeholder: 'search by keyword',
+                    })}
+                    id="service-type-ahead-input"
+                    className="usagov-search-autocomplete"
+                    name="query"
+                    type="text"
+                    ref="searchinput"
+                    required
+                  />
+                  {isOpen && inputValue.length >= 2 ? (
+                    <div className="dropdown" role="listbox">
+                      {this.state.suggestions.map((suggestion, index) => {
+                        return (
+                          <div
+                            key={suggestion}
+                            {...getItemProps({
+                              item: suggestion,
+                              className: this.optionClasses(
+                                index === highlightedIndex,
+                              ),
+                              role: 'option',
+                              'aria-selected': index === highlightedIndex,
+                            })}
+                            style={{
+                              fontWeight:
+                                selectedItem === suggestion ? 'bold' : 'normal',
+                            }}
+                          >
+                            {suggestion}
+                          </div>
+                        );
+                      })}
+                    </div>
+                  ) : null}
+                  <button type="submit" disabled={this.state.input.length <= 2}>
+                    <IconSearch color="#fff" />
+                    <span className="usa-sr-only">Search</span>
+                  </button>
+                </div>
+              </span>
+            </div>
+          )}
+        </Downshift>
+      </form>
+    );
+  }
+}
```

Comment:
The `<Downshift>` module is confusing, which is why I opened this PR - it at least gives us a starting point. The corresponding SCSS for this PR is mostly copied from another SCSS file, and I think it can be mostly removed in exchange for utility classes.